src/watertap_contrib/seto/multiperiod/VAGMD_batch_flowsheet.py

```diff
@@ -0,0 +1,455 @@
+        doc="Accumulated volume of distillate from previous time step",
+    )
+
+    m.fs.pre_acc_thermal_energy = Var(
+        initialize=0,
+        units=pyunits.kWh,
+        doc="Accumulated thermal energy consumption from previous time step",
+    )
+
+    m.fs.pre_acc_electric_energy = Var(
+        initialize=0,
+        units=pyunits.kWh,
+        doc="Accumulated electric energy consumption from previous time step",
+    )
+
+    m.fs.pre_acc_cooling_energy = Var(
+        initialize=0,
+        units=pyunits.kWh,
+        doc="Accumulated cooling energy consumption from previous time step",
+    )
+
+    m.fs.pre_thermal_power = Var(
+        initialize=10,
+        bounds=(0, None),
+        units=pyunits.kW,
+        doc="Thermal power consumption from previous time step",
+    )
+
+    m.fs.pre_cooling_power = Var(
+        initialize=10,
+        bounds=(0, None),
+        units=pyunits.kW,
+        doc="Cooling power consumption from previous time step",
+    )
+
+    m.fs.pre_cooling_pump_power_elec = Var(
+        initialize=10,
+        bounds=(0, None),
+        units=pyunits.kW,
+        doc="Electric power for pumping cooling water from previous time step",
+    )
+
+    m.fs.pre_feed_pump_power_elec = Var(
+        initialize=10,
+        bounds=(0, None),
+        units=pyunits.kW,
+        doc="Electric power for pumping feed water from previous time step",
+    )
+
+    """
+    Variables aggrated through periods
+    """
+    m.fs.acc_distillate_volume = Var(
+        initialize=0,
+        units=pyunits.L,
+        doc="Accumulated volume of distillate",
+    )
+
+    m.fs.acc_recovery_ratio = Var(
+        initialize=0,
+        units=pyunits.dimensionless,
+        doc="Accumulated recovery rate",
+    )
+
+    m.fs.acc_thermal_energy = Var(
+        initialize=0,
+        units=pyunits.kWh,
+        doc="Accumulated thermal energy consumption",
+    )
+
+    m.fs.acc_cooling_energy = Var(
+        initialize=0,
+        units=pyunits.kWh,
+        doc="Accumulated cooling energy consumption",
+    )
+
+    m.fs.acc_electric_energy = Var(
+        initialize=0,
+        units=pyunits.kWh,
+        doc="Accumulated electric energy consumption",
+    )
+
+    m.fs.specific_energy_consumption_thermal = Var(
+        initialize=100,
+        units=pyunits.kWh / pyunits.m**3,
+        doc="Specific thermal power consumption (kWh/m3)",
+    )
+
+    m.fs.specific_energy_consumption_electric = Var(
+        initialize=1,
+        units=pyunits.kWh / pyunits.m**3,
+        doc="Specific electric power consumption (kWh/m3)",
+    )
+
+    m.fs.gain_output_ratio = Var(
+        initialize=10,
+        units=pyunits.dimensionless,
+        doc="Gain output ratio",
+    )
+
+    """
+    Constriant equatiosn
+    """
+    mfs = m.fs
+
+    @mfs.Constraint(doc="Calculate time interval of each period")
+    def eq_dt(b):
+        if module_type == "AS7C1.5L":
+            return b.dt == 20352.55 / feed_flow_rate
```

Comment:
Where do these constants come from?

Reply:
I believe it comes from the datasheet of the membrane manufacturer Aquastill.